src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -121,7 +124,10 @@ static Replicator()
         /// <param name="config">The configuration to use to create the replicator</param>
         public Replicator([NotNull]ReplicatorConfiguration config)
         {
-            Config = CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(config), config).Freeze();
+            if (config?.Collections.Count <= 0)
+                throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, "Replicator configuration cannot be null.");
```

Comment:
It's hard to tell exactly what this is checking for.  Is it checking for empty collections or is it checking for null config?  If the config is null I think this will probably still throw NRE since config?.Collections will be null.  But there are two checks here, it seems....